### PR TITLE
MW-668/Create composite way logic

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/dataAccessLogic/DTOToPreviewConverter/wayDTOToWay.ts [hunk 1]
```diff
@@ -91,5 +91,8 @@ export const wayDTOToWay = (wayDTO: SchemasWayPopulatedResponse): Way => {
     }),
     createdAt: new Date(wayDTO.createdAt),
     lastUpdate: new Date(wayDTO.updatedAt),
+    children: wayDTO.children.map((child) => {
+      return wayDTOToWay(child);
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/DTOToPreviewConverter/wayDTOToWay.ts:95]: use short syntax

File: mw-webapp/src/logic/wayPage/wayStatistics/WayActiveStatistic.tsx [hunk 2]
```diff
@@ -176,22 +175,18 @@ export const WayActiveStatistic = (props: WayStatisticProps) => {
...
-    allJobsStatisticItem,
-    totalRecordsAmountStatisticItem,
     totalWayTimeStatisticItem,
+    totalRecordsAmountStatisticItem,
+    allJobsStatisticItem,
   ];
 
   const totalStatisticItemsSecondary = [
-    totalDaysOnWayStatisticItem,
+    // TotalDaysOnWayStatisticItem,
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/wayStatistics/WayActiveStatistic.tsx:184]: !

File: mw-webapp/src/logic/wayPage/wayStatistics/statisticBlock/statisticItem/StatisticItem.module.scss [hunk 3]
```diff
@@ -23,7 +23,7 @@ $secondaryStatisticItemPadding: 16px;
 }
 
 .secondary {
-  width: 105px;
+  width: 103px;
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/wayStatistics/statisticBlock/statisticItem/StatisticItem.module.scss:26]: 100?

File: mw-webapp/src/utils/createCompositeWay.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import {DayReport} from "src/model/businessModel/DayReport";
+import {Way} from "src/model/businessModel/Way";
+import {DateUtils} from "src/utils/DateUtils";
+
+/**
+ * Create composite way from array of ways
+ */
+export const createCompositeWay = (way: Way): Way => {
+  const allDayReports = way.children.flatMap((wayChild) => wayChild.dayReports);
+  const dayReportsSorted = allDayReports.sort((a, b) => b.createdAt.getTime() - a.createdAt.getTime());
```
Comment (tritonJS826) [inline mw-webapp/src/utils/createCompositeWay.ts:10]: do we need this sorting?

File: mw-webapp/src/utils/createCompositeWay.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      const updatedDayReport = new DayReport({
+        ...dayReport,
+        jobsDone: [...dayReport.jobsDone, ...dayReportSorted.jobsDone],
+        plans: [...dayReport.plans, ...dayReportSorted.plans],
+        problems: [...dayReport.problems, ...dayReportSorted.problems],
+        comments: [...dayReport.comments, ...dayReportSorted.comments],
+      });
+      dayReportsHashMap.set(DateUtils.getShortISODateValue(dayReportSorted.createdAt), updatedDayReport);
+    } else {
+      dayReportsHashMap.set(DateUtils.getShortISODateValue(dayReportSorted.createdAt), dayReportSorted);
```
Comment (tritonJS826) [inline mw-webapp/src/utils/createCompositeWay.ts:28]: duplicate

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/dataAccessLogic/DTOToPreviewConverter/wayDTOToWay.ts (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/src/dataAccessLogic/DTOToPreviewConverter/wayDTOToWay.ts]: Done

File: mw-webapp/src/logic/wayPage/wayStatistics/WayActiveStatistic.tsx (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/wayStatistics/WayActiveStatistic.tsx]: Deleted

File: mw-webapp/src/logic/wayPage/wayStatistics/statisticBlock/statisticItem/StatisticItem.module.scss (hunk 3 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/wayStatistics/statisticBlock/statisticItem/StatisticItem.module.scss]: Done

File: mw-webapp/src/utils/createCompositeWay.ts (hunk 4 above)
Comment (Ekaterina1994) [reply mw-webapp/src/utils/createCompositeWay.ts]: Deleted

**Pull request merged**